packages/@webex/plugin-meetings/src/recording-controller/index.ts

```diff
@@ -0,0 +1,315 @@
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private sessionId: string;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private locusUrl: string;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private locusId: string;
+
+  /**
+   * @param {MeetingRequest} request
+   * @param {Object} options
+   * @constructor
+   * @memberof RecordingController
+   */
+  constructor(
+    request: MeetingRequest,
+    options?: {
+      serviceUrl?: string;
+      sessionId: string;
+      locusUrl: string;
+      displayHints?: Array<string>;
+    }
+  ) {
+    this.initialize(request);
+    this.set(options);
+  }
+
+  /**
+   * @param {MeetingRequest} request
+   * @param {LocusInfo} info
+   * @returns {void}
+   * @private
+   * @memberof RecordingController
+   */
+  private initialize(request: MeetingRequest) {
+    this.request = request;
+  }
+
+  /**
+   * @param {Object} options
+   * @returns {void}
+   * @public
+   * @memberof RecordingController
+   */
+  public set(options?: {
+    serviceUrl?: string;
+    sessionId: string;
+    locusUrl: string;
+    displayHints?: Array<string>;
+  }) {
+    this.extract(options);
+  }
+
+  /**
+   * @param {string} url
+   * @returns {void}
+   * @public
+   * @memberof RecordingController
+   */
+  public setLocusUrl(url: string) {
+    this.locusUrl = url;
+    this.locusId = Util.extractLocusId(this.locusUrl);
+  }
+
+  /**
+   * @param {Array} hints
+   * @returns {void}
+   * @public
+   * @memberof RecordingController
+   */
+  public setDisplayHints(hints: Array<string>) {
+    this.displayHints = hints;
+  }
+
+  /**
+   * @param {string} id
+   * @returns {void}
+   * @public
+   * @memberof RecordingController
+   */
+  public setSessionId(id: string) {
+    this.sessionId = id;
+  }
+
+  /**
+   * @param {string} url
+   * @returns {void}
+   * @public
+   * @memberof RecordingController
+   */
+  public setServiceUrl(url: string) {
+    this.serviceUrl = url;
+  }
+
+  /**
+   * @returns {string}
+   * @public
+   * @memberof RecordingController
+   */
+  public getLocusUrl() {
+    return this.locusUrl;
+  }
+
+  /**
+   * @returns {string}
+   * @public
+   * @memberof RecordingController
+   */
+  public getLocusId() {
+    return this.locusId;
+  }
+
+  /**
+   * @returns {string}
+   * @public
+   * @memberof RecordingController
+   */
+  public getSessionId() {
+    return this.sessionId;
+  }
+
+  /**
+   * @returns {string}
+   * @public
+   * @memberof RecordingController
+   */
+  public getServiceUrl() {
+    return this.serviceUrl;
+  }
+
+  /**
+   * @returns {Array}
+   * @public
+   * @memberof RecordingController
+   */
+  public getDisplayHints() {
+    return this.displayHints;
+  }
+
+  /**
+   * @param {Object} options
+   * @returns {void}
+   * @private
+   * @memberof RecordingController
+   */
+  private extract(options?: {
```

Comment:
a very minor thing, but this method feels kind of redundant now. It's only called from `set()` and its body could just be inside `set()`